wiki/QFS-on-S3.md

```diff
@@ -0,0 +1,199 @@
+instead of pointing to an existing directory in chunkserver, it points to an hypothetical
+directory in which chunkservers can place S3 files. For example, the corresponding line
+for this parameter can read:
+``` 
+chunkServer.objectDir = s3://myfirstbucket.
+```
+In example, the part after "s3://" in the specified object directory, i.e. "myfirstbucket."
+(including the . sign at the end), is treated as the configuration suffix. Metaserver
+associates the configuration suffix with an S3 bucket name and AWS credentials to access
+the given bucket. Corresponding metaserver configuration changes are described in the next section.
+
+   This is the only required parameter that needs to be set up in chunkserver configuration file for S3 support. For the optional S3 parameters in chunkserver configuration file, please visit [here](https://github.com/quantcast/qfs/blob/master/conf/ChunkServer.prp#L290).
+
+   \* **One can specify multiple object directories with chunkServer.objectDir parameter.
+Please refer to [Specifying Multiple Object Store Directories](#MultipleObjectStoreSection) section in this guide for more details.**
+
+### metaserver configuration changes:
+1. *Add and set [metaServer.objectStoreEnabled](https://github.com/quantcast/qfs/blob/master/conf/MetaServer.prp#L1200) parameter.*
+
+   This parameter enables object store in metaserver. For example:
+```
+metaServer.objectStoreEnabled = 1
+```
+2. *For each configuration suffix specified in the chunkserver configuration file, specify
+the [chunkServer.diskQueue.&lt;object-store-directory-prefix&gt;bucketName](https://github.com/quantcast/qfs/blob/master/conf/MetaServer.prp#L1245),
+[chunkServer.diskQueue.&lt;object-store-directory-prefix&gt;accessKeyId](https://github.com/quantcast/qfs/blob/master/conf/MetaServer.prp#L1248),
+and [chunkServer.diskQueue.&lt;object-store-directory-prefix&gt;secretAccessKey](https://github.com/quantcast/qfs/blob/master/conf/MetaServer.prp#L1251)
+parameters.*
+
+   For example, using the same object directory specified in chunkserver configuration example, one can have:
+```
+metaServer.diskQueue.myfirstbucket.bucketName = <my-first-bucket-name> 
+metaServer.diskQueue.myfirstbucket.accessKeyId = <access-key-ID> 
+metaServer.diskQueue.myfirstbucket.secretAccessKey  = <secret-access-key> 
+```
+
+   The actual values for &lt;my-first-bucket-name&gt;, &lt;access-key-ID&gt; and
+&lt;secret-access-key&gt; depend on the existing S3 bucket and the IAM
+(Identity & Access Management) settings of the corresponding AWS account.
+
+3. *(optional) Add and set [metaServer.maxReplicasPerFile](https://github.com/quantcast/qfs/blob/master/conf/MetaServer.prp#L297)
+and [metaServer.maxReplicasPerRSFile](https://github.com/quantcast/qfs/blob/master/conf/MetaServer.prp#L302) parameters.*
+
+   These two parameters tell metaserver to store all files in S3 object store,
+regardless of the file creation arguments sent by QFS clients. For example:
+```
```

Comment:
These parameters are optional, and typically used in the case where QFS file system instance is configured with S3 storage only. The parameters override QFS clients file create parameters. It is possible to use mixed mode where the same QFS file system uses both S3 and disk store. In the later case these parameters should not be used.